cmd/titus-metadata-service/main.go

```diff
@@ -158,23 +189,43 @@ func main() {
 		/* Get the requisite configuration from environment variables */
 		listener := getListener(listenPort, listenerFd)
 
-		var err error
-		var container *titus.ContainerInfo
-		var signer *identity.Signer
-
-		if container, err = readTaskConfigFile(titusTaskInstanceID); err != nil {
-			log.Fatal(err)
+		mdscfg := types.MetadataServerConfiguration{
+			IAMARN:              iamARN,
+			TitusTaskInstanceID: titusTaskInstanceID,
+			Ipv4Address:         net.ParseIP(ipv4Address),
+			VpcID:               vpcID,
+			EniID:               eniID,
+			Region:              region,
+			Optimistic:          optimistic,
+			APIProtectEnabled:   apiProtectEnabled,
+		}
+		if parsedURL, err := url.Parse(backingMetadataServer); err == nil {
+			mdscfg.BackingMetadataServer = parsedURL
+		} else {
+			return cli.NewExitError(err.Error(), 1)
 		}
+
 		if metatronEnabled {
 			log.Info("Metatron enabled!")
-			if signer, err = identity.NewDefaultSigner(); err != nil {
-				log.Fatal(err)
+			if signer, err := identity.NewDefaultSigner(); err != nil {
+				log.WithError(err).Fatal("Cannot instantiate new default signer")
+				mdscfg.Signer = signer
+			}
+			if container, err := readTaskConfigFile(titusTaskInstanceID); err != nil {
```

Comment:
@rgulewich I hope it was okay to hoist this in here.

Reply:
Yeah - looks like I didn't push my local change that also did this, sorry. :(